components/sup/src/census.rs

```diff
@@ -144,10 +145,11 @@ impl CensusRing {
         // is an indeterminate health anywhere.
         service_rumors.with_keys(|(service_group, rumors)| {
             if let Ok(sg) = service_group_from_str(service_group) {
+                let local_member_id = Cow::from(&self.local_member_id);
```

Comment:
This is a bit tricky, with the original `or_insert` code, we allocated a new `CensusGroup` unconditionally. However, we can't just change to `or_insert_with` because the mutable borrow of self required to call [`entry`](http://doc.rust-lang.org/1.31.1/std/collections/hash_map/struct.HashMap.html#method.entry) conflicts with the immutable borrow to access `&self.local_member_id`. We could've unconditionally cloned `&self.local_member_id`, but using [`Cow`](http://doc.rust-lang.org/1.31.1/std/borrow/enum.Cow.html) gives us the ability to do the right thing at runtime.

Reply:
I friggin' love `Cow`s... they're so useful.

Reply:
🐮